modules/dkan_dataset_groups/dkan_dataset_groups.install

```diff
@@ -22,4 +22,49 @@ function dkan_dataset_groups_install() {
 function dkan_dataset_groups_disable() {
   //Remove field groups from the index.
   features_revert(array('dkan_sitewide_search_db' => array('search_api_index', 'search_api_server')));
-}
\ No newline at end of file
+}
+
+/**
+ * Implements hook_update_dependencies().
+ */
+function dkan_dataset_groups_update_dependencies() {
+
+  // We need to be sure datastore update runs before.
+  // dkan_datastore_status field it's added during that
+  // update. Without this dependecy dkan_dataset_groups_update_7000
+  // runs before throwing an entity metadata wrapper error.
+  $dependencies['dkan_dataset_groups'][7000] = array(
+    'dkan_datastore' => 7001,
+  );
+  return $dependencies;
+}
+
+/**
+ * Update resources without groups with the same group from parent dataset.
+ */
+function dkan_dataset_groups_update_7000(&$sandbox) {
+  if (!isset($sandbox['progress'])) {
+    $sandbox['progress'] = 0;
+    $sandbox['max'] = db_query("SELECT COUNT(DISTINCT fdr.field_dataset_ref_target_id) FROM {node} n LEFT JOIN {field_data_field_dataset_ref} fdr ON fdr.entity_id=n.nid LEFT JOIN {og_membership} og ON n.nid=og.etid WHERE n.type='resource' AND og.etid IS NULL AND fdr.field_dataset_ref_target_id IN (SELECT etid FROM {og_membership} WHERE etid IN (SELECT field_dataset_ref_target_id FROM {field_data_field_dataset_ref}))")->fetchField();
+  }
+
+  //Get all resources witout group but the parent dataset has groups
+  $result = db_query("SELECT fdr.field_dataset_ref_target_id FROM {node} n LEFT JOIN {field_data_field_dataset_ref fdr} ON fdr.entity_id=n.nid LEFT JOIN {og_membership} og ON n.nid=og.etid WHERE n.type='resource' AND og.etid IS NULL AND fdr.field_dataset_ref_target_id IN (SELECT etid FROM {og_membership} WHERE etid IN (SELECT field_dataset_ref_target_id FROM {field_data_field_dataset_ref})) LIMIT 0,10");
```

Comment:
We are wondering if this query will catch and correctly update resources that belong to two different datasets.